### PR TITLE
Add support for TFCE beta API

### PR DESCRIPTION
**Pull request by thrashr888**

### CODE REVIEW

**Comment by mwhooker** [comment]
would be great to get #81 in first

File: cost_estimate.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		if err != nil {
+			return nil, err
+		}
+
+		switch ce.Status {
+		case CostEstimateQueued:
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			case <-time.After(500 * time.Millisecond):
```
Comment (svanharmelen) [inline cost_estimate.go]: I think consistently polling every 500ms may be a bit too aggressive for the API. Maybe we should implement something like this instead: https://github.com/hashicorp/go-tfe/blob/master/logreader.go#L47

File: cost_estimate.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ CostEstimates = (*costEstimates)(nil)
+
+// CostEstimates describes all the costEstimate related methods that
+// the Terraform Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/enterprise/api/ (TBD)
```
[the remark this answers is not in the document]
Comment (thrashr888) [reply cost_estimate.go]: There's no tfce docs yet, unfortunately. Hopefully soon.

File: cost_estimate.go (hunk 1 above)
Comment (thrashr888) [reply cost_estimate.go]: I agree that 500 feels harsh but this worker is pretty quick and shouldn't take too many iterations to return. 500 matches the policy check interval, at least.

File: cost_estimate.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	CostEstimateFinished CostEstimateStatus = "finished"
+	CostEstimateQueued   CostEstimateStatus = "queued"
+)
+
+// CostEstimate represents a Terraform Enterprise costEstimate.
+type CostEstimate struct {
+	ID           string `jsonapi:"primary,cost-estimations"`
+	ErrorMessage string `jsonapi:"attr,error-message"`
+
+	ResourcesCount          string `jsonapi:"attr,resources-count"`
```
[the remark this answers is not in the document]
Comment (thrashr888) [reply cost_estimate.go]: None of the model structs are documented in-place. They just rely on the public API docs for now.

**Comment by davidcelis** [comment]
What's the plan for the resources themselves?

**Comment by thrashr888** [comment]
We're not ready to support the whole resources hash yet, given it might drift a bit. That and we don't have a particular need for it at the moment.

**Pull request merged**